docs/execute.rst

```diff
@@ -0,0 +1,171 @@
+available to control execution context. Note that ``path`` specifies
+in which folder to execute the notebook.
+
+**Execute/Run (preprocess)**: To actually run the notebook we call the method
+``preprocess``::
+
+    executor.execute()
+
+Hopefully, we will not get any errors during the notebook execution
+(see the last section for error handling).
+
+**Save**: Finally, save the resulting notebook with::
+
+    with open('executed_notebook.ipynb', 'w', encoding='utf-8') as f:
+        nbformat.write(nb, f)
+
+That's all. Your executed notebook will be saved in the current folder
+in the file ``executed_notebook.ipynb``.
+
+Execution arguments (traitlets)
+-------------------------------
+
+The arguments passed to :class:`Executor` are configuration options
+called `traitlets <https://traitlets.readthedocs.io/en/stable>`_.
+There are many cool things about traitlets. For example,
+they enforce the input type, and they can be accessed/modified as
+class attributes.
+
+Let's now discuss in more detail the two traitlets we used.
+
+The ``timeout`` traitlet defines the maximum time (in seconds) each notebook
+cell is allowed to run, if the execution takes longer an exception will be
+raised. The default is 30 s, so in cases of long-running cells you may want to
+specify an higher value. The ``timeout`` option can also be set to ``None``
+or ``-1`` to remove any restriction on execution time.
+
+The second traitlet, ``kernel_name``, allows specifying the name of the kernel
+to be used for the execution. By default, the kernel name is obtained from the
+notebook metadata. The traitlet ``kernel_name`` allows specifying a
+user-defined kernel, overriding the value in the notebook metadata. A common
+use case is that of a Python 2/3 library which includes documentation/testing
+notebooks. These notebooks will specify either a python2 or python3 kernel in
+their metadata (depending on the kernel used the last time the notebook was
+saved). In reality, these notebooks will work on both Python 2 and Python 3,
+and, for testing, it is important to be able to execute them programmatically
+on both versions. Here the traitlet ``kernel_name`` helps simplify and
+maintain consistency: we can just run a notebook twice, specifying first
+"python2" and then "python3" as the kernel name.
+
+Handling errors and exceptions
+------------------------------
+
+In the previous sections we saw how to save an executed notebook, assuming
+there are no execution errors. But, what if there are errors?
+
+Execution until first error
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+An error during the notebook execution, by default, will stop the execution
+and raise a ``CellExecutionError``. Conveniently, the source cell causing
+the error and the original error name and message are also printed.
+After an error, we can still save the notebook as before::
+
+    with open('executed_notebook.ipynb', mode='w', encoding='utf-8') as f:
```

Comment:
Isn't it also possible to just call `write` and a path to a notebook file, and it'll do all the `with open(` stuff under the hood?

Reply:
Yep, I blindly copied this section -- will update.